sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1340,7 +1340,10 @@ def need_ray_up(
             # Optimization: Try parse head ip from 'ray up' stdout.
             # Last line looks like: 'ssh ... <user>@<public head_ip>\n'
             position = stdout.rfind('@')
-            head_ip = stdout[position + 1:].strip()
+            # Use a regex to extract the IP address.
+            head_ip = re.search(
+                r'(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})',
+                stdout[position + 1:]).group(1)
             if not backend_utils.is_ip(head_ip):
                 # Something's wrong. Ok to not return a head_ip.
                 head_ip = None
```

Comment:
To make it more robust to abnormal cases, how about we use the following:
```suggestion
            # Use a regex to extract the IP address.
            ip_list = re.findall(
                backend_utils.IP_ADDR_REGEX,
                stdout[position + 1:])
            # If something's wrong. Ok to not return a head_ip.
            head_ip = None
            if len(ip_list) == 1:
                head_ip = ip_list[0]
```